tests/Doctrine/Tests/ORM/Mapping/TypedEnumFieldMapperTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Mapping;
+
+use Doctrine\ORM\Mapping\DefaultTypedFieldMapper;
+use Doctrine\ORM\Mapping\MappingException;
+use Doctrine\Tests\Models\Enums\FaultySwitch;
+use Doctrine\Tests\Models\Enums\SwitchStatus;
+use Doctrine\Tests\OrmTestCase;
+use ReflectionClass;
+
+use function sprintf;
+
+/**
+ * @requires PHP >= 8.1
+ */
+class TypedEnumFieldMapperTest extends OrmTestCase
+{
+    private static function defaultTypedFieldMapper(): DefaultTypedFieldMapper
+    {
+        return new DefaultTypedFieldMapper();
+    }
+
+    public function testNotBackedEnumThrows(): void
+    {
+        $reflectionClass = new ReflectionClass(FaultySwitch::class);
+
+        $this->expectException(MappingException::class);
+        $this->expectExceptionMessage(sprintf(
```

Comment:
I don't think you should use `sprintf` here. Please show the final exception message, as the user will see it instead.